src/Sentry/SentryFailedRequestHandler.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Sentry.Protocol;
+
+namespace Sentry
+{
+    internal class SentryFailedRequestHandler : ISentryFailedRequestHandler
+    {
+        private readonly IHub _hub;
+        private readonly SentryOptions? _options;
+
+        public string MechanismType { get => "SentryFailedRequestHandler"; }
+
+        /// <summary>
+        /// Initializes an instance of <see cref="SentryFailedRequestHandler"/>.
+        /// </summary>
+        internal SentryFailedRequestHandler(IHub hub, SentryOptions? options)
+        {
+            _hub = hub;
+            _options = options;
+        }
+
+        /// <summary>
+        /// Automaticlaly capture HTTP Client errors when captureFailedRequests
+        /// </summary>
+        /// <param name="request">The HttpRequestMessage sent</param>
+        /// <param name="response">The HttpResponse received</param>
+        public void CaptureEvent(HttpRequestMessage request, HttpResponseMessage response)
+        {
+            if (request is null)
+                return;
+
+            // Don't capture if the option is disabled
+            if (_options?.CaptureFailedRequests is false)
+                return;
+
+            // Don't capture events for successful requets
+            if (_options?.FailedRequestStatusCodes.Any(range => range.Contains(response.StatusCode)) is false)
+                return;
+
+            // Ignore requests to the Sentry DSN
+            var uri = request.RequestUri;
+            var requestString = uri?.OriginalString ?? "";
+            if (_options?.Dsn is { } dsn && new SubstringOrRegexPattern(dsn).IsMatch(requestString))
```

Comment:
Just compare the hosts.

```suggestion
            var uri = response.RequestMessage.RequestUri;
            if (_options.Dsn is { } dsn && new Uri(dsn).Host.Equals(uri.Host, StringComparison.OrdinalIgnoreCase))
```